tests/functional/auth_tests/test_key_pair.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from dbt.tests.util import run_dbt
+import pytest
+
+
+class TestKeyPairAuth:
```

Comment:
I have a feeling this test will fail because I doubt we have the private key in GHA.

Reply:
The secrets are added and this is passing now.